keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1010 @@
+
+This means that consumers can switch from one cluster manager to another without
+having to worry about compatibility issues or learning a new API. This
+can significantly reduce the learning curve and make it easier for
+consumers to adopt new tools.
+
+Moreover, the unified API can also provide a consistent user experience
+across different cluster managers. For example, if a consumer is used to a
+particular command or function in one tool, they can expect the same
+command or function to work in the same way in another tool. This can
+further enhance the usability and adoption of different cluster
+manager.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We try to summarize the most common properties that should be able to
+represent a cluster and support the above use case at the minimum scope.
+
+The target consumers of the API are users who manage the clusters and
+other tools to understand the clusters concept from the API for
+multicluster scheduling, workload distribution and cluster management.
+The API aims to provide the information for the consumers to answer the
+below questions:
+
+* Is the cluster under management?
+* How can I select a cluster?
+* Is the cluster healthy?
+* Does the cluster have certain capabilities or properties?
+* Does the cluster have sufficient resources?
+
+### Cluster Name
+
+It is required that cluster name is unique for each cluster, and it
+should also be unique among different providers (cluster manager). There
+are two option to ensure it is uniqueness
+
+#### Option 1
+
+The `metadata.name` of the cluster should never be set upon creation.
+Only `metadata.generateName` can be set.
+
+#### Option 2
+
+The `metadata.name` and `metadata.generateName` must have prefix which
+should be the same as the `spec.clusterManager.name`. Different cluster
+manager must set a different value of `spec.clusterManager.name` when
+the cluster is created.
+
+These two options can be achieved by a validating webhook or a
+`ValidatingAdmissionPolicy` resource.
+
+### Spec
+
+#### Display name
```

Comment:
Giving it more thought, Display name and ClusterManager should be part ObjectMeta.Name and ObjectMeta.Labels 
not part of the spec of ClusterInventory

Reply:
The good thing about this approach, is that we could leverage  `corev1.NodeSelectorTerm` and it's matcher methods. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3166

Reply:
metadata.name would have more strict validation and immutable. Display name on the other hand can be more flexible. 

I feel like ClusterManager is needed such that different provider can know which cluster they need to care under the case that multiple providers run in the same control plane. And having it in the spec field could allow stronger validations.